model/lists/Service/ListElementsUpdater.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
+ */
+
+namespace oat\taoBackOffice\model\lists\Service;
+
+use oat\tao\model\Lists\Business\Domain\Value;
+use oat\tao\model\Lists\Business\Domain\ValueCollection;
+use oat\tao\model\Lists\Business\Domain\ValueCollectionSearchRequest;
+use oat\tao\model\Lists\Business\Input\ValueCollectionSearchInput;
+use oat\tao\model\Lists\Business\Service\ValueCollectionService;
+use oat\taoBackOffice\model\lists\Contract\ListElementsUpdaterInterface;
+use core_kernel_classes_Class;
+use tao_helpers_Uri;
+
+class ListElementsUpdater implements ListElementsUpdaterInterface
+{
+    private const HUGE_LIST_MIN_ITEMS = 1000;
+    private const HUGE_LIST_MAX_MEMORY_MB = 500;
```

Comment:
Lets remove this constants and just update the elements in a batch of 50 (or a memory safer number) elements and that is it.

```suggestion
```

Reply:
I'm rechecking this, but I think that won't solve the issue. 

When adding each new item (`addElementToCollection`) we call `getValueByUriKey()` to check if the item is a new or existing one. In turn, `getValueByUriKey()` calls `ValueCollection::extractValueByUri()`, which relies on having all elements from the value collection already loaded in memory (it just iterates over `ValueCollection::$values` and compares the URIs).

So, if I'm right, we need to load all existing items in memory first, and when doing that for huge lists (1K+ items) we run out of memory. To make it work without actually loading all items, I think we would need to rework `RdfValueCollectionRepository::persist` (maybe `RdsValueCollectionRepository` too?) so it does not need to have all items loaded in memory to find duplicates.

Reply:
A quick check shows that if I don't fetch the current items (making `getListSearchInput` to return an input request with `limit=0`) then, when sending the form, existing items present in the form are always duplicated.

(Any limit value less than the list size limit (20K) would also be wrong and create duplicates)

Reply:
We need a bulk update with only key + value passed as argument (we receive this from the request).

Reply:
We are now just sending items that have been modified in the frontend to the backend, so I'm marking this htread as solved since there is no logic to dynamically raise the memory limit in the backend anymore.